src/main/java/soju/commands/DeadlineCommand.java

```diff
@@ -0,0 +1,59 @@
+package soju.commands;
+
+import java.time.LocalDate;
+
+import soju.SojuException;
+import soju.Storage;
+import soju.TaskList;
+import soju.Ui;
+import soju.tasks.Deadline;
+
+/**
+ * DeadlineCommand handles commands starting with deadline
+ */
+
+public class DeadlineCommand extends Command {
+    private Deadline deadlineTask;
+
+    /**
+     * Returns a DeadlineCommand
+     * @param input command input as a String
+     * @throws SojuException if invalid input
+     */
+    public DeadlineCommand(String input) throws SojuException {
+        try {
+            if (!input.startsWith("deadline ")) {
+                throw new SojuException("The description of a deadline cannot be empty.");
+            }
+
+            // Extract the part after "deadline "
+            String[] parts = input.substring(9).split(" /by ", 2);
```

Comment:
Perhaps you could have a clearer name instead of just parts?

Reply:
Agree, i think more descriptive variable names would make your code easier to read!